### PR TITLE
feat(web3hub): add search with mocks [LIVE-13187]

### PR DESCRIPTION
**Pull request by Justkant**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 
  - none screen behind a flag not enabled

### 📝 Description

Add search results with mocks
Add new test for search screen
Refactor with ViewModel hooks
Update CODEOWNERS folder name for Web3Hub

### ❓ Context

- **JIRA or GitHub link**: [LIVE-13187] [LIVE-13177]

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-13187]: https://ledgerhq.atlassian.net/browse/LIVE-13187?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LIVE-13177]: https://ledgerhq.atlassian.net/browse/LIVE-13177?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-mobile/src/newArch/features/Web3Hub/__integrations__/web3hub.integration.test.tsx [hunk 1]
```diff
@@ -113,4 +113,60 @@ describe("Web3Hub integration test", () => {
...
+    expect(await screen.findByText("Explore web3")).toBeOnTheScreen();
+
+    expect(await screen.findByTestId("web3hub-loading-indicator")).toBeOnTheScreen();
+    await waitForElementToBeRemoved(() => screen.getByTestId("web3hub-loading-indicator"), {
+      timeout: 1500, // timeout because we mock the request and fake 1s delay
+    });
+
+    expect(await screen.findByTestId("web3hub-main-header-search")).toBeOnTheScreen();
+    await user.press(screen.getByTestId("web3hub-main-header-search"));
+    expect(await screen.findByTestId("web3hub-search-header-search")).toBeOnTheScreen();
```
Comment (KVNLS) [inline apps/ledger-live-mobile/src/newArch/features/Web3Hub/__integrations__/web3hub.integration.test.tsx:129]: Don't we have other alternatives than *byTestId queries?
byTestId should be used as the last resort, is it the case here? https://callstack.github.io/react-native-testing-library/docs/guides/how-to-query#idiomatic-query-predicates

File: apps/ledger-live-mobile/src/newArch/features/Web3Hub/components/ManifestsList/useManifestsListViewModel.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { useInfiniteQuery } from "@tanstack/react-query";
+import {
+  fetchManifestsMock,
+  selectManifests,
+  getNextPageParam,
+} from "LLM/features/Web3Hub/utils/api/manifests";
+
+export default function useManifestListViewModel(selectedCategory: string) {
+  const manifestsQuery = useInfiniteQuery({
+    queryKey: ["web3hub/manifests", selectedCategory],
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-mobile/src/newArch/features/Web3Hub/components/ManifestsList/useManifestsListViewModel.ts:10]: I don't know if this is the only query you use, but `"web3hub/manifests"` could be part of an enum maybe as you might want to invalidate queries later?

(it's just code polishing 🫣)

File: apps/ledger-live-mobile/src/newArch/features/Web3Hub/__integrations__/web3hub.integration.test.tsx (hunk 1 above)
Comment (Justkant) [reply apps/ledger-live-mobile/src/newArch/features/Web3Hub/__integrations__/web3hub.integration.test.tsx:129]: I might try the placeholder but it was not working well

File: apps/ledger-live-mobile/src/newArch/features/Web3Hub/components/ManifestsList/useManifestsListViewModel.ts (hunk 2 above)
Comment (Justkant) [reply apps/ledger-live-mobile/src/newArch/features/Web3Hub/components/ManifestsList/useManifestsListViewModel.ts:10]: Yep usually I export the queryKey with the hook, I'll update

File: apps/ledger-live-mobile/src/newArch/features/Web3Hub/__integrations__/web3hub.integration.test.tsx (hunk 1 above)
Comment (KVNLS) [reply apps/ledger-live-mobile/src/newArch/features/Web3Hub/__integrations__/web3hub.integration.test.tsx]: I understand your point, but maybe we should assert that the navigation did well with an assertion of something that could be seen by the end user, wdyt?

Comment (Justkant) [reply apps/ledger-live-mobile/src/newArch/features/Web3Hub/__integrations__/web3hub.integration.test.tsx]: The search input is not the same and that's what I wanted to test, also it should focus the input but I have no way of testing that I think with RTL

And I already assert other changes on navigation to make sure I'm on the correct screen as a user would

**Review by KVNLS — APPROVED** [review]

**Review by mcayuelas-ledger — APPROVED** [review]

**Review by ComradeAERGO — APPROVED** [review]
Good for the current stage of development

**Pull request merged**